src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java

```diff
@@ -248,13 +258,17 @@ public Pipeline createPipelineAggregated(ItemGroup context, boolean showAggregat
             result.add(pipelineLatest);
             no--;
         }
-        
-        int pipelineCount = noOfPipelines;
-        if (pagingEnabled) {
-        	pipelineCount = firstProject.getBuilds().size();
+        int totalNoOfPipelines = firstProject.getBuilds().size();
+        component.setTotalNoOfPipelines(totalNoOfPipelines);
+        int startIndex = 0;
+        int retrieveSize = noOfPipelines;
+        if (pagingEnabled && !component.isFullScreenView()) {
+            startIndex = (component.getCurrentPage() - 1) * noOfPipelines;
+            retrieveSize = Math.min(totalNoOfPipelines - ((component.getCurrentPage() -1) * noOfPipelines), noOfPipelines);
         }
-        Iterator it = firstProject.getBuilds().iterator();
-        for (int i = 0; i < pipelineCount && it.hasNext(); i++) {
+
+        Iterator it = firstProject.getBuilds().listIterator(startIndex);
+        for (int i = startIndex; i < (startIndex + retrieveSize) && it.hasNext(); i++) {
```

Comment:
I would refactor it to: `for (AbstractBuild firstBuild : firstProject.getBuilds().subList(startIndex, startIndex + retrieveSize))`